src/main/java/blusunrize/immersiveengineering/common/blocks/generic/PostBlock.java

```diff
@@ -73,26 +76,29 @@ public PostBlock(String name, Properties blockProps)
 	@Override
 	public void onReplaced(@Nonnull BlockState state, World world, @Nonnull BlockPos pos, BlockState newState, boolean moving)
 	{
-		int dummyState = state.get(POST_SLAVE);
-		HorizontalOffset offset = state.get(HORIZONTAL_OFFSET);
-		if(dummyState > 0&&offset==HorizontalOffset.NONE)
-			world.setBlockState(pos.down(dummyState), Blocks.AIR.getDefaultState());
-		else if(dummyState==0)
+		if(state.getBlock()!=newState.getBlock())
 		{
-			spawnAsEntity(world, pos, new ItemStack(this));
-			final int highestBlock = 3;
-			BlockPos armStart = pos.up(highestBlock);
-			for(Direction d : Direction.BY_HORIZONTAL_INDEX)
+			int dummyState = state.get(POST_SLAVE);
+			HorizontalOffset offset = state.get(HORIZONTAL_OFFSET);
+			if(dummyState > 0&&offset==HorizontalOffset.NONE)
+				world.setBlockState(pos.down(dummyState), Blocks.AIR.getDefaultState());
+			else if(dummyState==0)
 			{
-				BlockPos armPos = armStart.offset(d);
-				BlockState armState = world.getBlockState(armPos);
-				if(armState.getBlock()==this&&armState.get(HORIZONTAL_OFFSET).getOffset().equals(d.getDirectionVec()))
-					world.setBlockState(armPos, Blocks.AIR.getDefaultState());
+				spawnAsEntity(world, pos, new ItemStack(this));
+				final int highestBlock = 3;
+				BlockPos armStart = pos.up(highestBlock);
+				for(Direction d : Direction.BY_HORIZONTAL_INDEX)
+				{
+					BlockPos armPos = armStart.offset(d);
+					BlockState armState = world.getBlockState(armPos);
+					if(armState.getBlock()==this&&armState.get(HORIZONTAL_OFFSET).getOffset().equals(d.getDirectionVec()))
+						world.setBlockState(armPos, Blocks.AIR.getDefaultState());
+				}
+				for(int i = 0; i <= highestBlock; ++i)
+					world.setBlockState(pos.up(i), Blocks.AIR.getDefaultState());
 			}
-			for(int i = 0; i <= highestBlock; ++i)
-				world.setBlockState(pos.up(i), Blocks.AIR.getDefaultState());
+			super.onReplaced(state, world, pos, newState, moving);
```

Comment:
Should this be inside the `if`?

Reply:
Doesn't *really* matter, because the method it overrides has another block != block check in it. It comes out to the same result